src/OSPSuite.Core/Domain/Services/ResidualCalculator.cs

```diff
@@ -99,20 +99,45 @@ private IReadOnlyList<Residual> calculateOutputResiduals(DataColumn simulationCo
          {
             var weight = outputMapping.Weight * outputMapping.WeightedObservedData.Weights[index];
             var observedValue = convertToBaseUnit(mergedDimension, currentObservedDataUnit, observedValueColumn[index]);
-            if (!observedValue.IsValid())
-               continue;
 
             var observedTime = observedTimeColumn[index];
-            var simulatedValue = simulationColumn.GetValue(observedTime);
-            if (!simulatedValue.IsValid())
+
+            if(simulationTimeIsTooShort(simulationColumn, observedTime))
+               continue;
+
+            if(observationShouldBeIgnored(observedValue))
                continue;
 
-            outputResiduals.Add(new Residual(observedTime, weight * residualCalculatorFunc(simulatedValue, observedValue, lloq), weight));
+            var simulatedValue = simulationColumn.GetValue(observedTime);
+
+            outputResiduals.Add(!simulatedValue.IsValid() ? 
+               new Residual(observedTime, residualValueForInvalidSimulationValue(simulatedValue), weight) : 
+               new Residual(observedTime, weight * residualCalculatorFunc(simulatedValue, observedValue, lloq), weight));
```

Comment:
matter of taste, but I would prefer `Condition ? Condition_True_Part : Condition_False_Part` rather than
`!Condition ? Condition_False_Part : Condition_True_Part`